lib/fastifySession.js

```diff
@@ -105,7 +105,12 @@ function fastifySession (fastify, options, next) {
         return
       }
 
-      request.session = restoredSession
+      const secureConnectionCookieMismatch = custom !== true &&
+        (typeof cookieOpts.secure === 'boolean'
+          ? cookieOpts.secure !== isConnectionSecure(request)
+          : session.cookie.secure !== isConnectionSecure(request))
+
+      request.session = secureConnectionCookieMismatch ? {} : restoredSession
```

Comment:
should it be an empty object and not a new `Session` instance?

Reply:
I dont know. I am open for discussion.

Reply:
If people do e.g. `request.session.save` this will fail now and https://github.com/fastify/session/blob/ed4f9ac14cca8c94f866aaa70fa63e5235fbaf33/types/types.d.ts#L14 is wrong